src/libs/OptionsListUtils.js

```diff
@@ -204,20 +205,58 @@ function hasReportDraftComment(report) {
         && lodashGet(reportsWithDraft, `${ONYXKEYS.COLLECTION.REPORTS_WITH_DRAFT}${report.reportID}`, false);
 }
 
+/**
+ * @param {Object} report
+ * @returns {Array<String>}
+ */
+function getParticipants(report) {
+    const participants = [...lodashGet(report, ['participants'], [])];
+    if (ReportUtils.isPolicyExpenseChat(report) && !_.contains(participants, report.ownerEmail)) {
+        participants.push(report.ownerEmail);
+    }
+    return participants;
+}
+
+/**
+ * Get the Avatar urls or return the icon according to the chat type
+ *
+ * @param {Object} report
+ * @returns {Array<*>}
+ */
+function getAvatarSources(report) {
```

Comment:
It seems like this might not be needed. Could we put all the report-type-specific `ReportUtils.isXYZ` checks in `OptionsListUtils.getReportIcons`? 

And if that's the case, this function would end up looking like:

```js
function getAvatarSources(report) {
    return _.map(lodashGet(report, 'icons', []), source => source || Expensicons.Profile);
}
```

Reply:
Ah I tried to implement the avatar sources logic in `getReportIcons` before but it's not possible because `getReportIcons` is called to update the reports to then save them into Onyx here:

https://github.com/Expensify/App/blob/daafe503149ff80c221a54d8df2b24082dcf9d42/src/libs/actions/PersonalDetails.js#L192-L211

And this approach was throwing errors because the icons (functional components) can't be saved in the storage.

I think we'll need a more holistic / different way to solve this (maybe we can return an object in `getReportIcons` to represent the icon and map that object to the corresponding Icon component just when the `Avatar` is rendered?) because also we're using an empty string to workaround this case in `getReportIcons`: https://github.com/Expensify/App/pull/7852/files#r814365077

I can create a ticket for that to discuss it deeper and address it later if it's ok :)

Reply:
Yep, thanks for the explanation. Sounds like  better idea to address in a follow-up. Can you please create the issue since you have more context and cc me?

Reply:
Also, if you don't think it's worth it we can just :donothing: and move on. I'll leave that up to you.

Reply:
This is the ticket https://github.com/Expensify/Expensify/issues/199856